translator/tools/polygon_tools.py

```diff
@@ -0,0 +1,258 @@
+
+        # Get left and right vectors
+        left = fwd_heading - 90
+        right = fwd_heading + 90
+
+        # Reset p1 to current point
+        p1 = geometry[i]
+
+        # get left and right points from direction and distance
+        origin = geopy.Point(p1[1], p1[0])
+        left_point = geodesic(
+            meters=polygon_width_in_meters/2).destination(origin, left)
+        right_point = geodesic(meters=polygon_width_in_meters /
+                               2).destination(origin, right)
+
+        # Append points to left and right lists
+        polygon_left_points.append([left_point.latitude, left_point.longitude])
+        polygon_right_points.append(
+            [right_point.latitude, right_point.longitude])
+
+    # Create list of points in correct order (all left points, then all right points in reverse order)
+    # This order is critical to prevent criss-crossing in the polygon
+    polygon_points = []
+
+    for i in polygon_left_points:
+        polygon_points.append(i)
+
+    for i in reversed(polygon_right_points):
+        polygon_points.append(i)
+
+    # Add first point again, to close polygon
+    polygon_points.append(polygon_left_points[0])
+
+    # Return generated polygon
+    polygon = Polygon(polygon_points)
+    return polygon
+
+
+# Check if point is in polygon
+def isPointInPolygon(point: Point, polygon: Polygon) -> bool:
+    """Determine if a point falls within a given polygon
+
+    Args:
+        point: Lat/long point
+        polygon: polygon
+
+    Returns:
+        Boolean of whether point is in polygon
+    """
+
+    if not point or not polygon or type(point) != Point or type(polygon) != Polygon:
+        return None
+    return polygon.contains(point)
+
+
+# function to get road direction by using geometry coordinates
+def get_road_direction_from_coordinates(coordinates):
+    if not coordinates or type(coordinates) != list or len(coordinates) < 2:
+        return None
+    try:
+        long_dif = coordinates[-1][0] - coordinates[0][0]
+        lat_dif = coordinates[-1][1] - coordinates[0][1]
+    except ValueError as e:
+        return None
+
+    if abs(long_dif) > abs(lat_dif):
+        if long_dif > 0:
+            direction = 'eastbound'
+        else:
+            direction = 'westbound'
+    elif lat_dif > 0:
+        direction = 'northbound'
+    else:
+        direction = 'southbound'
+
+    if lat_dif == 0 and long_dif == 0:
+        direction = None
+
+    return direction
+
+
+def angle_between_vectors_degrees(u, v):
+    """Return the angle between two vectors in any dimension space,
+    in degrees."""
+    return np.degrees(
+        math.acos(np.dot(u, v) / (np.linalg.norm(u) * np.linalg.norm(v))))
+
+
+def average_coordinates(coord1: list, coord2: list) -> list:
+    if len(coord1) != 2 or len(coord2) != 2:
+        return None
+    return [(coord1[0]+coord2[0])/2, (coord1[1]+coord2[1])/2]
+
+
+def average_polygon_to_centerline(polygon):
```

Comment:
It looks like `average_polygon_to_centerline` and `average_symmetric_polygon_to_centerline` are duplicates, can we combine?

Reply:
Good catch, I removed average_polygon_to_centerline